### ISSUE TITLE
log messages created using logger in pkg/logging are always categorized as severity INFO in GCP

### ISSUE THREAD

**Issue by lock14** [abcxyz/pkg#236]
### TL;DR

logs written with severity `ERROR` show as severity `INFO` when displayed in GCP log window.

See screen shot for example of the bug:
![image](https://github.com/abcxyz/pkg/assets/7461745/01b1c361-c5e4-4304-99ac-c028c3e256b5)

Even though the log message contains `"severity":"ERROR"`, GCP is categorizing it as severity `INFO`

### Expected behavior

I expected GCP to be able to parse the log message and correctly categorize the severity level

### Observed behavior

GCP categorizes all log messages as severity `INFO` when using the logger from `pkg/logging`

### Environment Details

```markdown
Observed bug behavior occurred when running an [Apache Beam](https://beam.apache.org/documentation/sdks/go/) pipeline via [GCP Dataflow](https://cloud.google.com/dataflow?hl=en).
```

**Comment by sethvargo** [comment]
Hi @lock14 - I am unable to reproduce this on Cloud Run:

![screenshot-20231102-Fdj373Cl@2x](https://github.com/abcxyz/pkg/assets/408570/869fc42d-0b7b-40d5-ac29-75c654000ea6)

Looking at the code for the beam logging, it [uses the standard logger](https://github.com/apache/beam/blob/cf48cafe48de1c5e9ba422c110214b4e6d7f69d1/sdks/go/pkg/beam/log/standard.go#L36), which [outputs to stderr by default](https://cs.opensource.google/go/go/+/refs/tags/go1.21.3:src/log/log.go;l=87). Google Cloud uses fluentd to capture logs, and the default configuration for fluentd is that all logs sent to stderr are classified as "ERROR". What happens if you use the beam logging library and output a debug, info, or warning message (you may need to change the level)? My suspicion is that it will also be logged at the "ERROR" level.

So why isn't pkg/logging working? If you read the ["example value" section in this table](https://cloud.google.com/logging/docs/structured-logging) closely, it says:

> Note: `message` is saved as `textPayload` if it is the only field remaining after the Logging agent moves the other special-purpose fields **and `detect_json` wasn't enabled**; otherwise message remains in `jsonPayload`. `detect_json` is not applicable to managed logging environments like Google Kubernetes Engine. If your log entry contains an exception stack trace, the exception stack trace should be set in this `message` JSON log field, so that the exception stack trace can be parsed and saved to Error Reporting.

My gut feeling is that the default Dataflow configuration does not set [`detect_json`](https://cloud.google.com/logging/docs/agent/logging/configuration#other-options) on the fluentd service on the underlying VMs. You [do not seem to be alone](https://stackoverflow.com/questions/55741432/how-to-send-and-filter-structured-logs-from-dataflow-job) in wanting this feature, and it was [requested by the Google Cloud Community](https://www.googlecloudcommunity.com/gc/Infrastructure-Compute-Storage/Batch-structured-logging/m-p/515421/highlight/true#M2302) in January 2023.

**Comment by lock14** [comment]
@sethvargo Thanks for looking into this. Curiously enough, `DEBUG`, `INFO`, `WARN`, and `ERROR` are all categorized correctly when using the beam logger.

![image](https://github.com/abcxyz/pkg/assets/7461745/7c86a4ad-4e9b-419a-b189-3274dcac8a5f)

**Comment by sethvargo** [comment]
I wonder if they are using a custom fluentd extractor then... I'll comment on https://github.com/abcxyz/github-metrics-aggregator/pull/171.

**Comment by sethvargo** [comment]
Hey @lock14 - we determined this was a shortcoming in Beam right? Like I don't think we can do anything in pkg unfortunately.

**Comment by lock14** [comment]
@sethvargo Yes, I will close this out as there is no action to take in pkg.

**Comment by lock14** [comment]
@sethvargo Yes, I will close this out as there is no action to take in pkg.
